### PR TITLE
Add io.guthub.dbchoco.muezzin

### PR DESCRIPTION
**Pull request by DBChoco**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/DBChoco/Muezzin**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ x Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.muezzin.app.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+build-options:
+  append-path: /usr/lib/sdk/node14/bin
+  cflags: -O2 -g
+  cxxflags: -O2 -g
```
Comment (hfiguiere) [inline com.muezzin.app.yml:19]:
```suggestion
```

this is not needed.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Neither the icon not the desktop files are installed.

Also it is missing an appstream file.

**Comment by DBChoco** [comment]
> Neither the icon not the desktop files are installed.
> 
> Also it is missing an appstream file.

Thanks for the feedback, but I don't know how to write/generate an appstream file, could you give me a link/example please ? Thanks again.

**Comment by DBChoco** [comment]
> > Neither the icon not the desktop files are installed.
> > Also it is missing an appstream file.
> 
> Thanks for the feedback, but I don't know how to write/generate an appstream file, could you give me a link/example please ? Thanks again.

My bad, just found here https://github.com/flathub/flathub/wiki/AppData-Guidelines
I'll work on it.

File: com.muezzin.app.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+       
+      - name: platform-bootstrap
+        buildsystem: simple
+        build-commands:
+          - |
+            set -e
+            mkdir -p /app/bin
+            install -Dm644 com.muezzin.app.png /app/share/icons/hicolor/128x128/apps/com.muezzin.app.png
+            install -Dm644 com.muezzin.app.desktop /app/share/applications/${FLATPAK_ID}.desktop
+            install -Dm644 $FLATPAK_ID.metainfo.xml /app/share/metainfo/$FLATPAK_ID.appdata.xml
```
Comment (hfiguiere) [inline com.muezzin.app.yml:66]:
```suggestion
            install -Dm644 com.muezzin-app.metainfo.xml.metainfo.xml /app/share/metainfo/$FLATPAK_ID.appdata.xml
```

File: com.muezzin.app.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../muezzin-run
+       
+      - name: platform-bootstrap
+        buildsystem: simple
+        build-commands:
+          - |
+            set -e
+            mkdir -p /app/bin
+            install -Dm644 com.muezzin.app.png /app/share/icons/hicolor/128x128/apps/com.muezzin.app.png
```
Comment (hfiguiere) [inline com.muezzin.app.yml:64]:
```suggestion
            install -Dm644 com.muezzin.app.png /app/share/icons/hicolor/128x128/apps/$FLATPAK_ID.png
```

File: com.muezzin.app.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
+app-id: com.muezzin.app
```
Comment (hfiguiere) [inline com.muezzin.app.yml:1]: this app-id doesn't follow the requirements.

1. the domain `muezzin.com` doesn't exist, so it's not in the control of the developers
2. `app` isn't an accetable application name.

See https://docs.flatpak.org/en/latest/conventions.html#application-ids

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
Please update the title of the PR to represent the app you are submitting. Thanks.

File: com.muezzin.app.yml (hunk 4 above)
Comment (DBChoco) [reply com.muezzin.app.yml:1]: Okay I'll change it to io.github.muezzin ; is that good ?

Comment (DBChoco) [reply com.muezzin.app.yml]: Oh didn't see this, I'll update it again.
Thanks!

**Comment by DBChoco** [comment]
Sorry for all the manual modifications, I had a problem with my git pull.
Is it enough like this ?

File: io.github.dbchoco.muezzin.xml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>io.github.muezzin</id>
```
Comment (hfiguiere) [inline io.github.dbchoco.muezzin.xml:3]:
```suggestion
  <id>io.github.dbchoco.muezzin</id>
```

File: io.github.dbchoco.muezzin.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>MIT</project_license>
+  
+  <description>
+    <p>
+      Islamic prayer times application that is highly customizable, free and open source.
+    </p>
+  </description>
+
+  <launchable type="desktop-id">io.github.muezzin.desktop</launchable>
```
Comment (hfiguiere) [inline io.github.dbchoco.muezzin.xml:19]:
```suggestion
  <launchable type="desktop-id">io.github.dbchoco.muezzin.desktop</launchable>
```

File: io.github.dbchoco.muezzin.yml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        build-commands:
+          - |
+            set -e
+            mkdir -p /app/bin
+            install -Dm644 io.github.dbchoco.muezzin.png /app/share/icons/hicolor/128x128/apps/$FLATPAK_ID.png
+            install -Dm644 io.github.dbchoco.muezzin.desktop /app/share/applications/${FLATPAK_ID}.desktop
+            install -Dm644 io.github.dbchoco.muezzin.metainfo.xml.metainfo.xml /app/share/metainfo/$FLATPAK_ID.appdata.xml
+        sources:
+          - type: file
+            path: io.github.dbchoco.muezzin.metainfo.xml
```
Comment (hfiguiere) [inline io.github.dbchoco.muezzin.yml:69]: the thing is this is not the actual file name in the repository (it should be renamed)

File: io.github.dbchoco.muezzin.desktop [hunk 8]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=Muezzin
+Exec=muezzin-run
+Terminal=false
+Type=Application
+Icon=io.github.muezzin
```
Comment (hfiguiere) [inline io.github.dbchoco.muezzin.desktop:6]:
```suggestion
Icon=io.github.dbchoco.muezzin
```

File: io.github.dbchoco.muezzin.yml [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+      - |
+        . ../flatpak-node/electron-builder-arch-args.sh
+        jq '.build.linux.target="dir"' <<<$(<package.json) > package.json
+        npm run dist -- $ELECTRON_BUILDER_ARCH_ARGS  --linux --dir
+      # Bundle app and dependencies
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../muezzin-run
+       
+      - name: platform-bootstrap
```
Comment (hfiguiere) [inline io.github.dbchoco.muezzin.yml:58]: this is incorrectly indented. It should be at the same level as line 21.

(it's yaml, it matters)

File: io.github.dbchoco.muezzin.yml [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        buildsystem: simple
+        build-commands:
+          - |
+            set -e
+            mkdir -p /app/bin
```
Comment (hfiguiere) [inline io.github.dbchoco.muezzin.yml:63]:
```suggestion
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by DBChoco** [comment]
Thanks again for everything.

**Comment by hfiguiere** [comment]
bot, build io.github.dbchoco.muezzin

File: io.github.dbchoco.muezzin.metainfo.xml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+                <li>Added Urdu language</li>
+            </ul>
+            <p>BugFixes</p>
+            <ul>
+                <li>Fixed time showing as 00:00 when 1 minute left.</li>
+                <li>Removed a few unnecessary modules</li>
+            </ul>
+        </description>
+      </release>
+  </releases>
```
Comment (hfiguiere) [inline io.github.dbchoco.muezzin.metainfo.xml:52]: `content_rating` is missing.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by DBChoco** [comment]
Hi there, sorry again, I just fixed 2 build issues, and it's building without a hitch on this side.
It should be okay to launch the build now.
Thanks!

**Comment by DBChoco** [comment]
bot, build io.github.dbchoco.muezzin

**Comment by DBChoco** [comment]
Wow I can do it too, I hope it doesn't bother you that I did that.

**Comment by DBChoco** [comment]
Fixed metainfo release date

**Comment by DBChoco** [comment]
There is an error I don't know how to fix:
`builddir/files/share/appdata/io.github.dbchoco.muezzin.appdata.xml: FAILED:
• attribute-missing     : <release> has no timestamp
Validation of files failed`

Could you help me please ?

**Comment by DBChoco** [comment]
> ?

`<release version="v0.1.7" date="2022/03/31" timestamp="1648829972">`
If I add the timestamp like this would this solve it ?

**Comment by hfiguiere** [comment]
> Wow I can do it too, I hope it doesn't bother you that I did that.

no actually it's better you trigger yourself.

**Comment by hfiguiere** [comment]
> > ?
> 
> `<release version="v0.1.7" date="2022/03/31" timestamp="1648829972">` If I add the timestamp like this would this solve it ?

date should be `2022-03-31` And no timestamp needed.

**Comment by DBChoco** [comment]
> 

Thanks a bunch

**Comment by DBChoco** [comment]
bot, build io.github.dbchoco.muezzin

**Comment by DBChoco** [comment]
Should I do something else now ?

**Comment by DBChoco** [comment]
Thanks a bunch everyone !

**Pull request closed without merging**